language-support/scala/codegen-testing/src/test/scala/com/digitalasset/ledger/client/binding/ValueSpec.scala

```diff
@@ -88,9 +88,16 @@ object ValueSpec {
       ValueCheck[P.Optional[A]](s"Option[$tName]")
     }
 
-    override def valueMap[A](implicit vc: ValueCheck[A]) = {
+    override def valueTextMap[A](implicit vc: ValueCheck[A]) = {
       import vc._
-      ValueCheck[P.Map[A]](s"Map[$tName]")
+      ValueCheck[P.TextMap[A]](s"Map[$tName]")
+    }
+
+    override def valueGenMap[K, V](implicit vcK: ValueCheck[K], vcV: ValueCheck[V]) = {
+      implicit def TA: Arbitrary[P.GenMap[K, V]] = implicitly[Arbitrary[P.GenMap[K, V]]]
+      implicit def TS: Shrink[P.GenMap[K, V]] = implicitly[Shrink[P.GenMap[K, V]]]
+      implicit def TV: Value[P.GenMap[K, V]] = implicitly[Value[P.GenMap[K, V]]]
```

Comment:
All of these are cyclic definitions. You've defined the `GenEncoding` and `ShrinkEncoding` GenMap overrides, so you can import `vc._` and derive the answers from those, such as is done for `valueParty` and `valueContractId`.

Reply:
fixed